contracts/ServiceNodeContribution.sol

```diff
@@ -184,20 +196,55 @@ contract ServiceNodeContribution is Shared {
     function cancelNode() public onlyOperator {
         require(!finalized, "Cannot cancel a finalized node.");
         require(!cancelled, "Node has already been cancelled.");
+        removeAndRefundContributor(msg.sender);
```

Comment:
It was intentional to leave the withdrawal of the contributors funds to themselves. Maybe its better for usability to have the operator cancelling automatically distribute their funds back to them. But not sure I like the added complexity brought by this

Reply:
This is still the case, we pass in the `msg.sender` which is the operator due to the `onlyOperator` modifier. The function removes the address that is passed into `removeAndRefundContributor`